File: algorithms/2315.count_asterisks/solution.go
package algorithms

// 2315. Count Asterisks (Easy)
// https://leetcode.com/problems/count-asterisks/description/

// Iterative solution
func countAsterisks(s string) int {
	numberOfAsterisks := 0
	currentPair := 0

	for _, char := range s {
		if char == '|' {
			currentPair++
		}

		if currentPair%2 != 0 {
			continue
		}

		if char == '*' {
			numberOfAsterisks++
		}
	}

	return numberOfAsterisks
}

File: algorithms/2315.count_asterisks/solution.ts
// 2315. Count Asterisks (Easy)
// https://leetcode.com/problems/count-asterisks/description/

// Iterative solution
const countAsterisks = (s: string): number => {
  const chars = [...s]
  let numberOfAsterisks = 0
  let currentPair = 0

  chars.forEach((char) => {
    if (char === "|") currentPair++
    if (currentPair % 2 !== 0) return
    if (char === "*") numberOfAsterisks++
  })

  return numberOfAsterisks
}
